File: apps/web/lib/services/notificationService.ts
import { apiFetch } from "./apiClient";
import type {
  NotificationListResponse,
  UnreadCountResponse,
} from "@/lib/types/notification";

export const notificationService = {
  list: (page = 1, size = 20) =>
    apiFetch<NotificationListResponse>(
      `/api/v1/notifications?page=${page}&size=${size}`,
    ),

  unreadCount: () =>
    apiFetch<UnreadCountResponse>("/api/v1/notifications/unread-count"),

  markRead: (id: string) =>
    apiFetch<void>(`/api/v1/notifications/${id}/read`, { method: "PATCH" }),

  markAllRead: () =>
    apiFetch<void>("/api/v1/notifications/read-all", { method: "POST" }),
};

File: apps/server/src/main/java/com/chtholly/notification/service/NotificationService.java
package com.chtholly.notification.service;

import com.chtholly.notification.api.dto.NotificationListResponse;
import com.chtholly.notification.api.dto.UnreadCountResponse;
import com.chtholly.notification.model.NotificationType;

import java.util.Map;

public interface NotificationService {

    NotificationListResponse list(long userId, int page, int size);

    UnreadCountResponse unreadCount(long userId);

    void markRead(long userId, long notificationId);

    void markAllRead(long userId);

    void create(long recipientUserId, NotificationType type, Map<String, Object> payload);
}
